### PR TITLE
docs: add new sample storage_configure_retries

### PR DESCRIPTION
**Pull request by BenWhitehead**
Outstanding items:
* [x] Libraries bom updated to a version which includes google-cloud-storage >= v2.2.0

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/storage/ConfigureRetries.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  static void deleteBlob(String bucketName, String blobName) {
+    // Update the retry settings
+    RetrySettings retrySettings =
+        StorageOptions.getDefaultRetrySettings()
+            .toBuilder()
+            // to set the max number of attempts to 10
+            .setMaxAttempts(10)
+            // to set the backoff multiplier to 3.0
+            .setRetryDelayMultiplier(3.0)
+            // to set the max duration of all attempts to 5 minutes
```
Comment (tritone) [inline samples/snippets/src/main/java/com/example/storage/ConfigureRetries.java:43]: Seems unrealistically long for a delete, but fine if this is what other langs have.

**Review by tritone — APPROVED** [review]
LGTM

File: samples/snippets/src/main/java/com/example/storage/ConfigureRetries.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    StorageOptions alwaysRetryStorageOptions =
+        StorageOptions.newBuilder()
+            // Configure our options so all requests will be retried even if they are
+            // non-idempotent.
+            .setStorageRetryStrategy(StorageRetryStrategy.getUniformStorageRetryStrategy())
+            // provide the previously configured retrySettings
+            .setRetrySettings(retrySettings)
+            .build();
+
+    // Instantiate a client with our options
```
Comment (cbonnie) [inline samples/snippets/src/main/java/com/example/storage/ConfigureRetries.java:56]: // Instantiate a client

**Review by cbonnie — COMMENTED** [review]
Left a few edits, otherwise LGTM.

File: samples/snippets/src/main/java/com/example/storage/ConfigureRetries.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+public final class ConfigureRetries {
+  public static void main(String[] args) {
+    String bucketName = "my-bucket";
+    String blobName = "blob/to/delete";
+    deleteBlob(bucketName, blobName);
+  }
+
+  static void deleteBlob(String bucketName, String blobName) {
+    // Update the retry settings
```
[the remark this answers is not in the document]
Comment (BenWhitehead) [reply samples/snippets/src/main/java/com/example/storage/ConfigureRetries.java:35]: I've fixed all of these over in another pr: https://github.com/googleapis/java-storage/pull/1188

**Pull request merged**